### ISSUE TITLE
Not Downloading

### ISSUE THREAD

**Issue by PK888888** [9-FS/2021-11-15-nHentai-to-PDF#1]
A little context, it worked before but today it just won't download anything. I get the usual 'Enter Holy Numbers" but now it's stuck at "Downloading...."

Let me know, thanks!

Edit: Grammar

**Comment by 9-FS** [comment]
Hello PK888888,
this is very likely due to cloudflare updating its protection. I can confirm that it also does not work for me anymore either. I'll try to get to it as soon as I can.

9FS

**Comment by PK888888** [comment]
Thanks for the prompt reply, no rush just curious if it was just me.

:)) Great tool btw HUGE props!

**Comment by 9-FS** [comment]
Hey @PK888888, it's been a while. I'm happy to announce that with today's patch (more like a complete rewrite), the downloader works again. 

Why did it take so long?
Basically, I used to scrape the gallery HTML for the image links and then download them. To circumvent cloudflare, my initial thought was to take that to the next level and simulate  a whole browser and a user clicking through the gallery to collect the image links. I managed to do that, but it was really unreliable and tended to just randomly crash. I kinda gave up at that point.
Things changed when I was made aware that nhentai provides a limited API a  couple of days ago. I found a way to access that without all the browser emulating overhead and can now directly download the images. 

Btw I also updated the whole project structure and coding style to my today's standard. It was very interesting to see how much I have changed within the last year.

Sorry for taking so long, but I hope you'll still find my tool useful. Have fun!
